src/sentry/static/sentry/app/components/onboardingWizard/task.tsx

```diff
@@ -72,7 +72,19 @@ function Task({router, task, onSkip, onMarkComplete, forwardedRef, organization}
     }
 
     if (task.actionType === 'app') {
-      navigateTo(`${task.location}?onboardingTask`, router);
+      // For the "second platform" task, it's confusing to continue linking to the "create new project"
+      // page after the user has created the new project -- instead, link to the install
+      // instructions for the corresponding project to prompt the user to send an event
+      // to the new project.
+      let location = task.location;
+      if (task.task === OnboardingTaskKey.SECOND_PLATFORM && task.status === 'pending') {
+        // if there's no project id assiciated with the task, display project selector prompt.
+        const projectSlug = task.project
+          ? organization.projects.find(project => project.id === `${task.project}`)!.slug
+          : ':projectId';
+        location = `/settings/${organization.slug}/projects/${projectSlug}/install/`;
+      }
+      navigateTo(`${location}?onboardingTask`, router);
```

Comment:
I don't like that the logic in this file, here and below (L151 - 156), is kind of a one-off -- it's checking for a single task+status combination and adding special logic for that particular circumstance. That makes the overall task logic harder to grok, and it makes it harder to understand how the task is constructed. But the only alternative I could come up with was adding special "pending_description" and "pending_location" attributes to all the tasks in the taskConfig file (or, worse, some kind of state --> description and location mapping), which seemed like overkill when there's only one task/status combination that needs this special handling.

If anyone has better suggestions for refactoring this, I'm all ears.

Reply:
I think this task's `actionType` should not be `app`, but rather it should be `action`. You'll get to abstract all this logic into a function under `task.action`. 

From a cursory glance, it seems to me that "Adding a second platform" is as much of an `action` as "Invite your team members".